xbmc/addons/BinaryAddonCache.h

```diff
@@ -31,8 +31,9 @@ namespace ADDON {
 class CBinaryAddonCache : public Observer
 {
 public:
-  virtual ~CBinaryAddonCache();
+  virtual ~CBinaryAddonCache() { }
```

Comment:
if you declare 
```virtual ~CBinaryAddonCache();```
in the header and 
``` 
CBinaryAddonCache::~CBinaryAddonCache()
{
  // no code
}
``` in the cpp it works fine and segfault gone.